src/components/protocol_handler/test/protocol_handler_tm_test.cc

```diff
@@ -3367,11 +3482,872 @@ TEST_F(ProtocolHandlerImplTest, SendMessageToMobileApp_SendMultiframeMessage) {
   times++;
 
   // Act
+  protocol_handler_impl->SendMessageToMobileApp(message, kFinalMessage);
+
+  EXPECT_TRUE(waiter->WaitFor(times, kAsyncExpectationsTimeout));
+}
+
+TEST_F(ProtocolHandlerImplTest,
+       AddProtocolObserver_NullObserver_CountNotChanged) {
+  ASSERT_EQ(0u, protocol_handler_impl->get_protocol_observers_count());
+
+  protocol_handler_impl->AddProtocolObserver(NULL);
+  EXPECT_EQ(0u, protocol_handler_impl->get_protocol_observers_count());
+}
+
+TEST_F(ProtocolHandlerImplTest, AddProtocolObserver_CorrectData_CountIncrease) {
+  ASSERT_EQ(0u, protocol_handler_impl->get_protocol_observers_count());
+
+  protocol_handler_impl->AddProtocolObserver(&protocol_observer_mock);
+  EXPECT_EQ(1u, protocol_handler_impl->get_protocol_observers_count());
+}
+
+TEST_F(ProtocolHandlerImplTest,
+       RemoveProtocolObserver_NullObserver_CountNotChanged) {
+  ASSERT_EQ(0u, protocol_handler_impl->get_protocol_observers_count());
+
+  protocol_handler_impl->AddProtocolObserver(&protocol_observer_mock);
+  ASSERT_EQ(1u, protocol_handler_impl->get_protocol_observers_count());
+
+  protocol_handler_impl->RemoveProtocolObserver(NULL);
+  EXPECT_EQ(1u, protocol_handler_impl->get_protocol_observers_count());
+}
+
+TEST_F(ProtocolHandlerImplTest,
+       RemoveProtocolObserver_CorrectData_CountDecrease) {
+  ASSERT_EQ(0u, protocol_handler_impl->get_protocol_observers_count());
+
+  protocol_handler_impl->AddProtocolObserver(&protocol_observer_mock);
+  ASSERT_EQ(1u, protocol_handler_impl->get_protocol_observers_count());
+
+  protocol_handler_impl->RemoveProtocolObserver(&protocol_observer_mock);
+  EXPECT_EQ(0u, protocol_handler_impl->get_protocol_observers_count());
+}
+
+TEST_F(ProtocolHandlerImplTest, GetSessionObserver_CorrectObserver) {
```

Comment:
I think that this test is redundant, code of this method is too simple to check.